protobufs/livekit_rtc_v2.proto

```diff
@@ -0,0 +1,156 @@
+// Copyright 2023 LiveKit, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package livekit;
+option go_package = "github.com/livekit/protocol/livekit";
+option csharp_namespace = "LiveKit.Proto";
+option ruby_package = "LiveKit::Proto";
+
+import "livekit_rtc.proto";
+import "livekit_models.proto";
+
+message Signalv2ClientMessage {
+  // monotonically increasing number starting at some non-zero number (should not rollover)
+  uint32 message_id = 1;
+
+  // last remote message processed
+  uint32 last_processed_remote_message_id = 2;
+
+  oneof message {
+    ConnectRequest connect_request = 3;
+    ReconnectRequest reconnect_request = 4;
+    AudioTrack audio_track = 5;
+    VideoTrack video_track = 6;
+    SessionDescription publisher_sdp = 7;
+    SessionDescription subscriber_sdp = 8;
+    TrickleRequest ice_candidate = 9;
+    MuteTrackRequest mute_track_request = 10;
+    UpdateSubscription update_subscription = 11;
+    UpdateTrackSettings update_track_settings = 12;
+    LeaveRequest leave_request = 13;
+    SubscriptionPermission update_subscription_permission = 14;
+    SimulateScenario simulate_scenario = 15;
+    UpdateParticipantMetadata update_participant_metadata = 16;
+    Ping ping = 17;
+    UpdateLocalAudioTrack update_local_audio_track = 18;
+    UpdateLocalVideoTrack update_local_video_track = 19;
+  }
+}
+
+message Signalv2ClientEnvelope {
+  // collection of client -> server messages
+  repeated Signalv2ClientMessage client_messages = 1;
+}
+
+message Signalv2ServerMessage {
+  // monotonically increasing number starting at some non-zero number (should not rollover)
+  uint32 message_id = 1;
+
+  // last remote message processed
+  uint32 last_processed_remote_message_id = 2;
+
+  repeated ErrorResponse errors = 3;
```

Comment:
Introducing this to send errors in processing specific messages from client. Informational only (maybe we can make it actionable down the line). Thinking about this to give clients a better view of any errors on server side.

NOTE: Individual messages can include `request_id` to get a response (irrespective of an error). That mechanism will continue to be supported. This is a generic mechanism to just send down errors processing any message.

Reply:
Having second thoughts about this. Not sure if this would be useful, i. e. sending back error description for specific messages (keyed by message id).

I am on the wall between deciding to give it a slot as a common entity and not using it  OR removing it now and later adding a common entity with a slot number that is high. I think I am overthinking this. Leaning towards removing it and add it later if needed.

Please let me know if you have thoughts.

Reply:
I'm not a fan of this field. 
We wanted to make rpc style requests more explicit anyways and we, also, have the envelope now that allows multiple responses being sent down, couldn't we just have the error as part of the oneof message of `Signalv2ServerMessage` then?